### PR TITLE
On ufs-community/ccpp-physics Issue #172 

### PR DESCRIPTION
**Pull request by ShanSunNOAA**
-- 'tisfc' is no longer modified by the atmospheric model, i.e., allowing the ice model to determine the ice temperature over both sea ice and lake ice; 
-- Changing tisfc from intent(inout) to intent(in).

### CODE REVIEW

**Review by Qingfu-Liu — APPROVED** [review]

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90 [hunk 1]
```diff
@@ -111,7 +110,6 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, iopt_lake, iopt_l
               if (cice(i) >= min_lakeice) then
                 icy(i)    = .true.
                 islmsk(i) = 2
-                tisfc(i)  = max(timin, min(tisfc(i), tgice))
```
Comment (mzhangw) [inline physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90:114]: Could you confirm if the intent of tisfc for sea ice will also be applied to lake ice?

Comment (ShanSunNOAA) [reply physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90:114]: Mike Barlage confirmed that it should be the lake ice model to determine the ice temperature, not here in the atmospheric model. 

Shan

**Comment by yangfanglin** [comment]
Shan,  could you please attach to this PR the  slides you and/or Lydia made to illustrate the problem the PR is addressing ?

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90 (hunk 1 above)
Comment (mzhangw) [reply physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90:114]: As the PoC of the CLM lake model, @tanyasmirnova Could you confirm that?

**Comment by ShanSunNOAA** [comment]
This issue is identified [here](https://docs.google.com/presentation/d/1jod3YnSht0ceWPKWkDmV1Y82C3y4C4ekCql5Dmp3gU8/edit?usp=sharing). The output from my experiment is [here](https://docs.google.com/presentation/d/11Wg4aw25tPAuZrVzTd_GQycPKaThii6gJla18rASxJY/edit?usp=sharing).

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90 (hunk 1 above)
Comment (tanyasmirnova) [reply physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90:114]: @mzhangw The CLM model sends ice temperature into tisfc(i) variable. However, the line you point to should not be used with the CLM model. The testing of the CLM model has been done with frac_grid = .false., therefore the specific for the CLM lake model code is in the part of the code when frac_land=.false.:

186             else ! Not ocean and not land
187               **is_clm** = lkm>0 .and. iopt_lake==iopt_lake_clm .and. use_lake_model(i)>0
188               if (cice(i) >= min_lakeice) then
189                 icy(i) = .true.
190                 **if(.not.is_clm) then**
191                   tisfc(i) = max(timin, min(tisfc(i), tgice))
192                 endif
193                 islmsk(i) = 2
194               else
195                 cice(i)   = zero
196                 hice(i)   = zero
197                 islmsk(i) = 0
198                 icy(i)    = .false.
199               endif
200               islmsk_cice(i) = islmsk(i)
201               flag_cice(i)   = .false.
202               **if(is_clm) then**
203                 wet(i) = .true.
204                 if (icy(i)) then
205                   tsfco(i) = max(tisfc(i), tgice)
206                 endif
207               else if(cice(i) < one) then
208                 wet(i) = .true. ! some open lake
209                 if (icy(i)) then
210                   tsfco(i) = max(tisfc(i), tgice)
211                 endif
212               endif
213             endif

**Review by yangfanglin — APPROVED** [review]
Shan had discussed these changes with a few EMC developers before opening this PR.  Approved.

**Comment by grantfirl** [comment]
@ShanSunNOAA Have you run UFS regression tests with this change? Do you expect any baseline changes?

**Comment by ShanSunNOAA** [comment]
I haven't run any regression tests. I'd be happy to do so, when it is my turn for the PR. I expect most baselines, if not all, will change.

**Comment by grantfirl** [comment]
> I haven't run any regression tests. I'd be happy to do so, when it is my turn for the PR. I expect most baselines, if not all, will change.

UFS code managers expect RTs to be run before it is your turn in the queue. Their job is to verify that the code is behaving as you claim, so you need to show proof of running RTs on Hera/Derecho/Hercules via a log that shows which tests fail and those failures need to be expected/explainable from your code changes.

I've started PRs for fv3atm and ufs-weather-model (https://github.com/NOAA-EMC/fv3atm/pull/783, https://github.com/ufs-community/ufs-weather-model/pull/2137). I'll fill out the templates and ask for help if I need it.

**Comment by ShanSunNOAA** [comment]
Sounds good. Thanks for the info and the heads up.

File: physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90 (hunk 1 above)
Comment (grantfirl) [reply physics/Interstitials/UFS_SCM_NEPTUNE/GFS_surface_composites_pre.F90:114]: @mzhangw @tanyasmirnova Following up on this discussion, so the changes in this PR are OK and don't need to be modified for CLM, correct?

**Review by grantfirl — APPROVED** [review]

**Comment by grantfirl** [comment]
@ShanSunNOAA Can you please pull in the latest ufs/dev branch code into your PR branch? Let me know if you need help with this.

**Comment by FernandoAndrade-NOAA** [comment]
Testing for #[2137](https://github.com/ufs-community/ufs-weather-model/pull/2137) has completed successfully, please continue with the merge process.

**Pull request merged**